src/game.py

```diff
@@ -3,34 +3,47 @@
 from src.player import Player
 
 class Game():
+    EXIT = "exit"
 
     def __init__(self, cli=Cli(), board=Board(), players=[Player(token="X"),
                                                           Player(token="O")]):
         self.cli = cli
         self.board = board
         self.players = players
+        self.exit = False
 
     def start(self):
         self.cli.welcome()
         self.play()
 
     def play(self):
         self.cli.display_board(self.board)
-        while not self.board.full():
+        while not self.game_is_over():
             self.turn()
+        self.handle_exit()
 
-        self.cli.goodbye()
-    
-    def turn(self, error=False):
-        if error: self.cli.invalid_move()
-
-        player_choice = self.cli.prompt_player_turn(self.current_player())
-        move = self.input_to_index(player_choice)
-        if self.board.valid_move(move):
+    def handle_exit(self):
+        if self.exit:
+            self.cli.handle_exit()
+        else:
+            self.cli.handle_game_end()
+
+    def game_is_over(self):
+        return self.exit or self.board.full()
+
+    def turn(self):
+        while True:
+            player_choice = self.cli.prompt_player_turn(self.current_player())
+            if player_choice.lower() == self.EXIT:
```

Comment:
Seems odd that the game should know to .lower() the return value of player_choice.

Reply:
I agree, it felt a bit awkward to me as well, I've adjusted it